plugins/cost-insights/src/components/BarChart/BarChart.tsx

```diff
@@ -20,51 +20,59 @@ import {
   BarChart as RechartsBarChart,
   CartesianGrid,
   ContentRenderer,
-  TooltipProps,
+  TooltipProps as RechartsTooltipProps,
+  RechartsFunction,
   ResponsiveContainer,
   Tooltip as RechartsTooltip,
   XAxis,
   YAxis,
-  TooltipPayload,
 } from 'recharts';
 import { Box, useTheme } from '@material-ui/core';
 import { BarChartTick } from './BarChartTick';
 import { BarChartStepper } from './BarChartStepper';
-import { Tooltip, TooltipItemProps } from '../Tooltip';
-
 import { currencyFormatter } from '../../utils/formatters';
 import {
   BarChartData,
-  Maybe,
   ResourceData,
   DataKey,
   CostInsightsTheme,
 } from '../../types';
-import { notEmpty } from '../../utils/assert';
 import { useBarChartStyles } from '../../utils/styles';
 import { resourceSort } from '../../utils/sort';
 
 export type BarChartProps = {
+  resources: ResourceData[];
   responsive?: boolean;
   displayAmount?: number;
-  barChartData: BarChartData;
-  getTooltipItem: (payload: TooltipPayload) => Maybe<TooltipItemProps>;
-  resources: ResourceData[];
+  options?: Partial<BarChartData>;
+  tooltip?: ContentRenderer<RechartsTooltipProps>;
+  onClick?: RechartsFunction;
+  onMouseMove?: RechartsFunction;
 };
 
 export const BarChart = ({
+  resources,
   responsive = true,
   displayAmount = 6,
-  barChartData,
-  getTooltipItem,
-  resources,
+  options = {},
+  tooltip,
+  onClick,
+  onMouseMove,
```

Comment:
One of the benefits of lifting the tooltip out of `BarChart` and passing it via props, is that the props for `BarChart` are specific to the component itself.

The event handlers on the recharts `BarChart` are exposed to allow parent components to gain access to child component state. This allows for different types of tooltips to be implemented for the same `BarChart` component.